librz/core/cmd/cmd_eval.c

```diff
@@ -162,32 +127,46 @@ RZ_API RZ_OWN RzList /*<char *>*/ *rz_core_theme_list(RZ_NONNULL RzCore *core) {
 		RZ_FREE(path);
 	}
 
-	RzList *list = rz_list_newf(free);
-	rz_list_append(list, strdup("default"));
-	ht_su_foreach(themes, dict2keylist, list);
-
-	rz_list_sort(list, (RzListComparator)strcmp, NULL);
-	ht_su_free(themes);
-	return list;
+	RzPVector *vec = set_s_to_vector(themes);
+	if (!vec) {
+		set_s_free(themes);
+		return NULL;
+	}
+	rz_pvector_push(vec, strdup("default"));
+	rz_pvector_sort(vec, (RzPVectorComparator)compare_strings, NULL);
+	set_s_free(themes);
+	return vec;
 }
 
 RZ_API void rz_core_theme_nextpal(RzCore *core, RzConsPalSeekMode mode) {
-	RzListIter *iter;
-	const char *fn;
-	RzList *files = rz_core_theme_list(core);
-
-	rz_list_foreach (files, iter, fn) {
-		if (*fn && *fn != '.') {
-			if (!pal_seek(core, mode, fn, iter)) {
-				goto done;
+	rz_return_if_fail(core && core->curtheme);
+
+	void **iter;
+	size_t idx;
+	RzPVector *files = rz_core_theme_list(core);
+	rz_pvector_enumerate (files, iter, idx) {
+		const char *fn = *iter;
+		if (strcmp(fn, core->curtheme)) {
+			continue;
+		}
+		if (mode == RZ_CONS_PAL_SEEK_PREVIOUS) {
+			if (idx == 0) {
+				break;
 			}
+			fn = rz_pvector_at(files, idx - 1);
+		} else if (mode == RZ_CONS_PAL_SEEK_NEXT) {
+			if (idx == rz_pvector_len(files) - 1) {
```

Comment:
```suggestion
			if (idx >= rz_pvector_len(files) - 1) {
```

Reply:
`rz_pvector_enumerate` iterates from `0` to `rz_pvector_len(pvec) - 1`